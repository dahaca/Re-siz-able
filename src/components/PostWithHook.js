import React from 'react'
import styled from 'styled-components'
import useResizer from '../lib/Resizable/useResizer'
import { PostContainer, Name, Close, Description } from './SocialPost'

const Arrow = styled.div`
  font-size: 3rem;
  user-select: none;
  position: absolute;
  bottom: 1rem;

  :hover {
    color: #87cefa;
  }

  :active {
    cursor: grabbing;
  }
`

const EwArrow = styled(Arrow)`
  right: 7rem;
  cursor: ew-resize;
`

const NsArrow = styled(Arrow)`
  left: 1rem;
  cursor: ns-resize;
  writing-mode: vertical-lr;
`

const NwseArrow = styled(Arrow)`
  right: 0;
  cursor: nwse-resize;
  transform: rotate(45deg);
`

// Uncomment the default values and remove the ref for an alternative way of using useResizer()!
const resizerConfig = {
  minHeight: 270,
  minWidth: 300,
  maxWidth: 800,
  preserveRatio: true,
  // defaultWidth: 400,
  // defaultHeight: 400,
}

const PostWithHook = () => {
  const { elementRef, width, height, handleMouseDown } = useResizer(
    resizerConfig
  )

  return (
    <PostContainer ref={elementRef} style={{ width: width, height: height }}>
      <Name>John Doe</Name>
      <Close />
      <Description>
        That's a functional component using the useResizer hook!
      </Description>
      <NsArrow onMouseDown={handleMouseDown('bottom')}>{'<=>'}</NsArrow>
      <EwArrow onMouseDown={handleMouseDown('right')}>{'<=>'}</EwArrow>
      <NwseArrow onMouseDown={handleMouseDown('bottom-right')}>
        {'<=>'}
      </NwseArrow>
    </PostContainer>
  )
}

export default PostWithHook
