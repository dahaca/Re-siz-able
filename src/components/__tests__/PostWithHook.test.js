import React from 'react'
import { render, mount } from 'enzyme'
import PostWithHook from '../PostWithHook'

describe('Test PostWithHook', () => {
  it('should render correctly', () => {
    expect(render(<PostWithHook />)).toMatchSnapshot()
  })

  it('should change width by 20px', () => {
    const post = mount(<PostWithHook />)
    const rightHandle = post.find('EwArrow')
    rightHandle.simulate('click')
  })
})
