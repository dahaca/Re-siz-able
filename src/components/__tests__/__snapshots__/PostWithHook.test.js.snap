// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PostWithHook should render correctly 1`] = `
<div
  class="sc-htpNat cCdKpA"
>
  <div
    class="sc-bxivhb cgeqyC"
  >
    John Doe
  </div>
  <div
    class="sc-ifAKCX lgNMH"
  />
  <div
    class="sc-EHOje jCbNlh"
  >
    That's a functional component using the useResizer hook!
  </div>
  <div
    class="sc-bZQynM sc-htoDjs fYJJur"
  >
    &lt;=&gt;
  </div>
  <div
    class="sc-bZQynM sc-gzVnrw cGdEkI"
  >
    &lt;=&gt;
  </div>
  <div
    class="sc-bZQynM sc-dnqmqq ddxEqp"
  >
    &lt;=&gt;
  </div>
</div>
`;
